sdk/core/AzureCore/Tests/UserAgentPolicyTests.swift

```diff
@@ -29,29 +29,242 @@ import XCTest
 
 class UserAgentPolicyTests: XCTestCase {
 
-    /// Test that the default user agent policy is properly applied.
-    func testDefaultUserAgentPolicy() {
-        // This is an example of a functional test case.
-        // Use XCTAssert and related functions to verify your tests produce the correct results.
+    /// Test that the user agent policy creates the correct user agent with the empty initializer
+    func test_UserAgentPolicy_WithEmptyInitializer() {
         let policy = UserAgentPolicy()
         let userAgent = policy.userAgent
-        XCTAssertEqual(userAgent, "ios/5.0 (iPhone) AzureCore/0.1.0")
+        XCTAssertEqual(userAgent, "azsdk-ios")
     }
 
-    /// Test that supplying a baseUserAgent without overwrite still captures much of the standard data.
-    func testCustomUserAgentPolicyNoOverwrite() {
-        let policy = UserAgentPolicy(baseUserAgent: "MyAgent")
+    /// Test that the user agent policy creates the correct user agent with the custom useragent initializer
+    func test_UserAgentPolicy_WithCustomUserAgentInitializer() {
+        let customAgent = "MyAgent/5.0"
+        let policy = UserAgentPolicy(userAgent: customAgent)
         let userAgent = policy.userAgent
-        XCTAssertEqual(userAgent, "MyAgent ios/5.0 (iPhone) AzureCore/0.1.0")
+        XCTAssertEqual(userAgent, customAgent)
```

Comment:
So a custom string hijacks the entire UA string? What if someone wanted to simply append to the standard string? I thought that was a supported use case in Python? Even if it's not, I remember getting queries from partner teams that wanted to do just that.

Reply:
Yeah, so that's how it was previously, I just wrote a test case to cover it. Basically, to do the use case you suggest, first the SDK client would construct its initial pipeline with the useragent policy:

```swift
policies: [
    ...
    UserAgentPolicy(sdkName: "TheSDK",  sdkVersion: "1.0", ...)
    ...
]
```

Later, the consumer / extender / subclasser would insert their own user agent policy ahead of the default one:

```swift
client.insert(policy: UserAgentPolicy(userAgent: "MyServiceAgent"), at: .front)
```

Because the custom user agent policy is executed first, it will set the user agent to the explicitly provided string. Then, when the client's default user agent policy executes, it will see that there's an existing user agent and it will prepend the value with the conventional user agent string (see `test_UserAgentPolicy_WithCurrentNonDefaultUserAgent_PrependsHeaderValue`)

Reply:
Sweet. Sounds reasonable.

Reply:
Personally I could be convinced to remove the initializer that just blanket sets the user agent. In fact I think I'm going to do that. It offers no advantage over adding a HeadersPolicy that just sets the user agent:

```swift
let headers = HTTPHeaders()
headers[.userAgent] = "MyServiceAgent"
client.insert(policy: HeadersPolicy(addingHeaders: headers), at: .front)
```

Reply:
Removed this initializer in 353a31e and fixed the tests in 94739a1